Extend/Payment_Gateway.php

```diff
@@ -104,70 +106,72 @@ public function executePayment($dAmount, &$oOrder)
                 $sSeverity = ucfirst($oStatus->getSeverity());
                 $sCode = $oStatus->getCode();
                 $sDescription = $oStatus->getDescription();
-                $this->oLogger->error("\t$sSeverity with code $sCode and message '$sDescription' occurred.");
+                $this->_oLogger->error("\t$sSeverity with code $sCode and message '$sDescription' occurred.");
             }
             return false;
         }
         $sPageUrl = null;
         if ($oResponse instanceof InteractionResponse) {
             $sPageUrl = $oResponse->getRedirectUrl();
         }
+
+        $this->_addTransactionToDb($oResponse, $oOrder);
+
         Registry::getUtils()->redirect($sPageUrl);
         return true;
     }
 
     /**
      * Returns country code
      *
-     * @param string $countryId
+     * @param string $sCountryId
      *
      * @return string
      *
      * @SuppressWarnings(PHPMD.Coverage)
      */
-    private static function _getCountryCode($countryId)
+    private function _getCountryCode(string $sCountryId): string
```

Comment:
why remove the `static`?

Reply:
we don't use it anywhere else when it isn't needed